### PR TITLE
Add unit tests + refacto

### PR DESCRIPTION
**Pull request by adelplace**
The goal here is to provide a pretty full coverage with unit tests in order to create a first stable version of the library.

Feel free to discuss about this work ;)

### CODE REVIEW

File: src/Apps.php [hunk 1]
```diff
@@ -2,15 +2,30 @@
 
 namespace OneSignal;
 
-use Symfony\Component\OptionsResolver\OptionsResolver;
+use OneSignal\Resolver\ResolverFactory;
 
 class Apps
 {
+    /**
+     * @var OneSignal
```
Comment (norkunas) [inline src/Apps.php]: Revert this

**Comment by norkunas** [comment]
For v1 this will be good, but for v2 I think it's worth to drop the options resolvers and just implement payload objects (DTO's), increase php version to 7.1 and use strict types

**Comment by adelplace** [comment]
Hi,

Thanks for the review ;)
Concerning the comments on attributes I wouldn't say they're unecessary as they are recognized by IDEs, they help for autocompletion. Anyway I removed the one you pointed, tell me if it's not enough.
About the idea to implement playload objects, I totally agree. It 's a way more user friendly.

**Comment by norkunas** [comment]
IDE's are smart enough to recognize the provided objects because the constructor has the same typehints, this only helps where constructor parameters are not objects, so I was trying to say that other docs could also be removed :)

**Comment by adelplace** [comment]
I have to admit that you're right, it should be better now.

File: src/Apps.php [hunk 2]
```diff
@@ -2,15 +2,24 @@
...
+use OneSignal\Resolver\ResolverFactory;
 
 class Apps
 {
     protected $api;
 
-    public function __construct(OneSignal $api)
+    private $resolverFactory;
+
+    /**
```
Comment (norkunas) [inline src/Apps.php]: Remove this doc

File: src/Notifications.php [hunk 3]
```diff
@@ -2,18 +2,26 @@
...
 class Notifications
 {
     const NOTIFICATIONS_LIMIT = 50;
 
     protected $api;
 
-    public function __construct(OneSignal $api)
+    private $resolverFactory;
+
+    /**
```
Comment (norkunas) [inline src/Notifications.php]: Remove this doc

File: src/Resolver/DeviceResolver.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    /**
+     * @return bool
+     */
+    public function getIsNewDevice()
+    {
+        return $this->isNewDevice;
+    }
+
+    /**
+     * @param $isNewDevice
```
Comment (norkunas) [inline src/Resolver/DeviceResolver.php]: should be `@param bool $isNewDevice`

File: src/Resolver/DeviceResolver.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+     *
+     * @return $this
+     */
+    public function setIsNewDevice($isNewDevice)
+    {
+        $this->isNewDevice = $isNewDevice;
+
+        return $this;
```
Comment (norkunas) [inline src/Resolver/DeviceResolver.php]: This is not fluent api so this should be removed

File: src/Resolver/ResolverFactory.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+namespace OneSignal\Resolver;
+
+use OneSignal\Config;
+
+class ResolverFactory
+{
+    private $config;
+
+    /**
+     * ResolverFactory constructor.
```
Comment (norkunas) [inline src/Resolver/ResolverFactory.php]: Remove this doc

**Comment by norkunas** [comment]
Hi, will review again this week and if ok will release v1

**Comment by norkunas** [comment]
Thank you @adelplace

**Comment by adelplace** [comment]
You're welcome !

**Pull request merged**